modules/react/collection/stories/Collection.stories.mdx

```diff
@@ -53,13 +55,42 @@ item for selection, maintaining a cursor, or anything else.
 ### Dynamic Items
 
 The `ListBox` also handles a dynamic collection of items. Instead of providing each `ListBox.Item`
-statically, provide a render function instead. The function is called with an `items` value that is
-the same was what's provided to the model. By default, providing items will enable virtualization.
-This example adds a `maxHeight` to ensure overflow. Virtualization uses absolute positioning of each
-item, which could cause problems for popup menus. If your item count is low, pass
-`shouldVirtualize={false}` to disable virtualization.
+statically, provide a render function instead. The function is called with an `item` value that is
+the same as what's provided to the model.
 
-<ExampleCodeBlock code={DynamicItems} />
+#### Registering Items
+
+Dynamic items can be passed in as an array of strings or objects via the `items` prop or the model
+config.
+
+##### Array of Strings
+
+Use strings when your data is a simple list of options. Each string value is used as the unique
+identifer.
+
+<ExampleCodeBlock code={DynamicItemsStrings} />
+
+##### Array of Objects
+
+Use objects when your data contains additional information. Each object must have an `id` to ensure
+it's registered and searchable. Additional information such as icons or if the item is disabled can
+be included in the object and will be accessible in the render function.
+
+<ExampleCodeBlock code={DynamicItemsObjects} />
+
+> **Note: The `id` should match your text value for each item. This ensures proper selection of
+> items in components like `Select` and `Autocomplete`. If your `id` is different from your text
+> value, you'll have to add a
+> [translation layer](https://codesandbox.io/s/combobox-object-items-forked-mc7tm4). **
+
+#### Virtualization
+
+By default, providing items dynamically will enable virtualization. This example adds a `maxHeight`
+to ensure overflow. Virtualization uses absolute positioning of each item, which could cause
+problems for popup menus. If your item count is low, pass `shouldVirtualize={false}` to disable
+virtualization.
```

Comment:
>  By default, providing items will enable virtualization. This example adds a `maxHeight` to ensure overflow. Virtualization uses absolute positioning of each item, which could cause problems for popup menus

Menus default virtualization to `false`.

Also, what issues? It might be useful to describe what issues may be encountered. For example, an HTML container element can render to the width of children, but if the children are not always in the DOM, the min width may not be calculated according to the widest item. A `width` or `minWidth` should be set on virtualized lists.